lib/ContentTypeParser.js

```diff
@@ -1,67 +1,174 @@
+/* eslint-disable no-useless-return */
 'use strict'
 
+const lru = require('tiny-lru')
+
 function ContentTypeParser () {
   this.customParsers = {}
-  this.parserList = []
+  this.customParsers['application/json'] = defaultJsonParser
+  this.parserList = ['application/json']
+  this.cache = lru(100)
 }
 
-ContentTypeParser.prototype.add = function (contentType, fn) {
+ContentTypeParser.prototype.add = function (contentType, opts, parser) {
   if (typeof contentType !== 'string') throw new Error('The content type should be a string')
   if (contentType.length === 0) throw new Error('The content type cannot be an empty string')
-  if (typeof fn !== 'function') throw new Error('The content type handler should be a function')
+  if (typeof parser !== 'function') throw new Error('The content type handler should be a function')
 
   if (this.hasParser(contentType)) {
     throw new Error(`Content type parser '${contentType}' already present.`)
   }
+
+  parser.asString = opts.asString === true
+  parser.asBuffer = opts.asBuffer === true
```

Comment:
what is that `parser` object? Is it a function? I would prefer if we didn't attach properties to it, but we rather wrap it in its own object with these properties.

Also, I would prefer we had a single option in the opts instead so we don't have a problem to validate these.

Reply:
> what is that parser object? Is it a function? 

A function

> I would prefer if we didn't attach properties to it, but we rather wrap it in its own object with these properties.

Ok, we can do that

> Also, I would prefer we had a single option in the opts instead so we don't have a problem to validate these.

I'm not sure I have understood, can you rephrase, maybe with an example?

Reply:
```js
{ parseAs: 'string' }
```

Which translated to boolean flags. But in this way it's clear to the user only a single flag is possible.

Reply:
Done